lib/puppet/util.rb

```diff
@@ -25,26 +25,49 @@ module Util
 
   extend Puppet::Util::SymbolicFileMode
 
-  # Run some code with a specific environment.  Resets the environment back to
-  # what it was at the end of the code.
-  # Windows can store unicode chars in the environment as keys or values, but
-  # Rubys ENV tries to roundtrip them through the local codepage, which can
-  # cause encoding problems - underlying helpers use Windows APIs on Windows
-  # see https://bugs.ruby-lang.org/issues/8822
-  def self.withenv(hash, mode = :posix)
```

Comment:
Any reason this is not deprecated? I don't see it behind a private switch so IMHO it must be deprecated and removed on a major version bump.